subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java

```diff
@@ -363,30 +364,23 @@ private void nagUserAboutUnreadableInputsOrOutputs(String propertyType, String p
         }
         LOGGER.info("Cannot access {} property '{}' of {}", propertyType, propertyName, getDisplayName(), cause);
         boolean isDestinationDir = propertyName.equals("destinationDir");
+        DeprecationMessageBuilder<?> builder;
         if (isDestinationDir && task instanceof Copy) {
-            DeprecationLogger.deprecateAction("Cannot access a file in the destination directory (see --info log for details). Copying to a directory which contains unreadable content")
-                .withAdvice("Use the method Copy.ignoreExistingContentInDestinationDir().")
-                .willBecomeAnErrorInGradle8()
-                // TODO: Document
-                .undocumented()
-                .nagUser();
+            builder = DeprecationLogger.deprecateAction("Cannot access a file in the destination directory (see --info log for details). Copying to a directory which contains unreadable content")
+                .withAdvice("Use the method Copy.ignoreExistingContentInDestinationDir().");
         } else if (isDestinationDir && task instanceof Sync) {
-            DeprecationLogger.deprecateAction("Cannot access a file in the destination directory (see --info log for details). Syncing to a directory which contains unreadable content")
-                .withAdvice("Use a Copy task with Copy.ignoreExistingContentInDestinationDir() instead.")
-                .willBecomeAnErrorInGradle8()
-                // TODO: Document
-                .undocumented()
-                .nagUser();
+            builder = DeprecationLogger.deprecateAction("Cannot access a file in the destination directory (see --info log for details). Syncing to a directory which contains unreadable content")
+                .withAdvice("Use a Copy task with Copy.ignoreExistingContentInDestinationDir() instead.");
         } else {
-            DeprecationLogger.deprecateAction(String.format("Cannot access %s property '%s' of %s (see --info log for details). Accessing unreadable inputs or outputs",
+            builder = DeprecationLogger.deprecateAction(String.format("Cannot access %s property '%s' of %s (see --info log for details). Accessing unreadable inputs or outputs",
                     propertyType, propertyName, getDisplayName()))
-                .withAdvice("Declare the property as untracked.")
-                .willBecomeAnErrorInGradle8()
-                // TODO: Document
-                .undocumented()
-                .nagUser();
+                .withAdvice("Declare the property as untracked.");
 
         }
+        builder
+            .willBecomeAnErrorInGradle8()
+            .withUpgradeGuideSection(7, "deprecate_unreadable_output")
```

Comment:
Not all of these are about outputs, should we link to someplace different?

Reply:
I renamed the link to `declare_unreadable_input_output`. I don't think we need separate sections for input or output properties.